appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -206,12 +207,21 @@ public void reset(ContextHandle contextHandle) {
     }
 
     private boolean isApplicationEnabled(String appId) {
+        boolean result = false;
         if (appId != null) {
             Application app = applications.getApplication(appId);
-            if (app != null)
-                return deployment.isAppEnabled(app);
+            if (app != null) {
+                result = deployment.isAppEnabled(app);
+            } else { 
+                // if app is null then it is likely that appId is still deploying
+                // and its enabled status has not been written to the domain.xml yet
+                // this can happen for example with a Startup EJB submitting something
+                // it its startup method. Reference Payara GitHub issue 204              
+                logger.info("Job submitted for " + appId + " likely during deployment. Continuing...");
+                result = true;
```

Comment:
Will dig some more to see if there's an alternative mechanism to determine app deployment status. It isn't too fatal making the assumption as the task will just fail if there's some classloader or somesuch problem.